### DISCUSSION TITLE
How do you manage macOS .plists?

### DISCUSSION THREAD

**Discussion by BasixKOR** [twpayne/chezmoi#2965] [General]
I'm trying to manage some of my macOS application settings with chezmoi, but I've come across some hurdles:

- macOS plists are binary files, which makes it hard to work when templating is needed (secrets, conditionals, etc.)
- Many applications (ab)use the Preferences plists to store their temporary, machine-specific state, which I'd rather not sync.

How are you managing your macOS preferences? I'd love to see how others are working with this.

**Comment by Jaharmi** [comment]
I’m just getting started with Chezmoi, but macOS property list files — at least the ones people tend to care about in places like `~/Library/Preferences` — are essentially an on-disk cache of in-memory settings.

For best results, they should only be accessed/edited with a CFPreferences-aware tool. The `defaults` command on macOS is one. There are libraries that use the CFPreferences framework for various scripting/programming languages. This should result in the modification of the _setting_ (including in-memory), and not just _the on-disk artifact_ we see as plists.

One example I’ve found for Chezmoi is [a script in twpayne’s dotfiles](https://github.com/twpayne/dotfiles/blob/9dbfd6816b0ad23e85ac9cd369444eeb2f8e5a2a/home/.chezmoiscripts/darwin/run_onchange_after_configure.sh).
